### PR TITLE
[SW-1103] Add custom DDS profile for slow image publishing issues

### PR DESCRIPTION
**Pull request by tcappellari-bdai**
## Change Overview

Added `custom_profile.xml` with a custom DDS profile that Mich made to help with slow image publishing issues. 

Did not solve the issue completely but it raised mine and Mich's frequency from ~5 Hz to ~15 Hz (on ethernet + laptop). Katie's rate also increased but was overall lower than ours both with and without the custom profile. 

Also updated the README with this update.

## Testing Done

Please create a checklist of tests you plan to do and check off the ones that have been completed successfully. Ensure that ROS 2 tests use `domain_coordinator` to prevent port conflicts. Further guidance for testing can be found on the [ros utilities wiki](https://github.com/bdaiinstitute/ros_utilities/wiki/Testing-guidelines).

### CODE REVIEW

**Comment by tcappellari-bdai** [comment]
* **#400** <a href="https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/400?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tcappellari-bdai and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: custom_profile.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            </transport_descriptor>
+        </transport_descriptors>
+        <participant profile_name="default" is_default_profile="true">
+            <rtps>
+                <!--
+                    Set participant transports
+                -->
+                <userTransports>
+                    <transport_id>udp_transport</transport_id>
+                    <transport_id>shm_transport</transport_id>
```
Comment (amessing-bdai) [inline custom_profile.xml:23]: Can you add comments with links to documentation on these?

File: custom_profile.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<dds xmlns="http://www.eprosima.com">
+    <profiles>
+        <transport_descriptors>
+            <transport_descriptor>
+                <transport_id>udp_transport</transport_id>
+                <type>UDPv4</type>
+            </transport_descriptor>
+            <transport_descriptor>
+                <transport_id>shm_transport</transport_id>
```
Comment (bhung-bdai) [inline custom_profile.xml:10]: Huh. Didn't realized shared memory works across the python publishing but cool

**Review by bhung-bdai — APPROVED** [review]

File: custom_profile.xml
Comment (khughes-bdai) [inline custom_profile.xml:1]: Nit: can we give this a slightly more descriptive name? Maybe `dds_profile.xml` or something

**Review by khughes-bdai — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -97,6 +97,14 @@ The driver can publish both compressed images (under `/<Robot Name>/camera/<came
 
 The driver also has the option to publish a stitched image created from Spot's front left and front right cameras (similar to what is seen on the tablet). If you wish to enable this, launch the driver with `stitch_front_images:=True`, and the image will be published under `/<Robot Name>/camera/frontmiddle_virtual/image`. In order to receive meaningful stitched images, you will have to specify the parameters `virtual_camera_intrinsics`, `virtual_camera_projection_plane`, `virtual_camera_plane_distance`, and `stitched_image_row_padding` (see [`spot_driver/config/spot_ros_example.yaml`](spot_driver/config/spot_ros_example.yaml) for some default values). 
 
+> **_NOTE:_**  
+If your image publishing rate is very slow, you can try 
+> - connecting to your robot via ethernet cable 
+> - exporting a custom DDS profile we have provided by running the following in the same terminal your driver will run in:.
```
Comment (khughes-bdai) [inline README.md:103]:
```suggestion
> - exporting a custom DDS profile we have provided by running the following in the same terminal your driver will run in, or adding to your `.bashrc`:.
```

File: custom_profile.xml (hunk 1 above)
Comment (tcappellari-bdai) [reply custom_profile.xml]: ya i can ask mich if he has any he referenced

File: custom_profile.xml (hunk 2 above)
Comment (amessing-bdai) [reply custom_profile.xml]: I don't think it does, but the image processing pipeline is all c++ components so that is sped up

**Review by bhung-bdai — APPROVED** [review]

File: custom_profile.xml (hunk 2 above)
Comment (mhidalgo-bdai) [reply custom_profile.xml]: FWIW you do get shared memory transport in Python too, but you can't escape the conversion from Python messages to C messages and back.

**Pull request merged**